src/WarcraftLegacies.Source/Quests/Zandalar/QuestZandalarOutpost.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Zandalar
+{
+  /// <summary>
+  /// The Trolls can establish an outpost in the Echo Isles.
+  /// </summary>
+  public sealed class QuestZandalarOutpost : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+    private readonly ObjectiveAnyUnitInRect _enterZandalarRegion;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestZandalar"/>.
+    /// </summary>
+    public QuestZandalarOutpost() : base("Zandalar Outpost",
+      "The trolls of the echo islands have said we can set up an outpost on one of the islands.",
```

Comment:
```suggestion
      "It has been an age since the height of Zandalar's glory. Since then, our Darkspear cousins have grown strong nestled in the Horde's breast. We should establish an outpost in the Echo Isles, so that Zandalari and Darkspear may meet on shared ground.",
```
Your RewardFlavour and RewardDescription are awesome, well done.